### ISSUE TITLE
Error upgrading from 2.x to 3.x: The destination vertex must be a vertex record

### ISSUE THREAD

**Issue by adepase** [orientechnologies/orientdb#8669]
### Java Version: "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, 
mixed mode)
## Expected behavior  
Until (at least) version 2.2.23
Directly from Studio->Browse, execute the following query:

```
SELECT percorso from (
	SELECT shortestPath(son, parent, 'OUT', 'hypernym') as percorso from (
		select from (
			select list(son) as son, list(parent) as parent from (
				SELECT EXPAND( $c ) 
					LET $a = ( select null as son, @rid as parent from synset where lemmas lucene 'animale' and lang='it'), 
              				$b = ( select @rid as son, null as parent from synset where lemmas lucene 'gatto' and lang='it' and verified is null ), 
					$c = UNIONALL( $a, $b )
                        )
                  ) unwind son, parent
	)
)
```

Result
```
PROPERTIES
--

#21:0 #22:0 #23:0 #24:0 #21:1 #22:1 ..More(2)
#21:2 #21:0 #22:0 #23:0 #24:0 #21:1 ..More(3)

```

## Actual behavior  
java.lang.IllegalArgumentException: The destinationVertex must be a vertex record

## Steps to reproduce  
See above. You can use the reduced example db in #8565 
I cannot find any upgrade instruction about changes in the sql that could be impacted in the above query.
Please, help, I cannot upgrade in these conditions

**Comment by luigidellaquila** [comment]
Hi @adepase 

Are you sure this think worked in v 2.2? I'm asking because shortestPath() function is the same, and it accepts two vertices as "from" and "to" 
In this example you are passing two lists, so I'd say a failure is the expected behavior...

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Oh, I think I got it, could you please do a test for me: try to change the "parent" alias with something else (eg. "parent_"), it's probably just a misbehaviour with reserved words 

Thanks

Luigi

**Comment by adepase** [comment]
Sure, you got it! It works (even when incapsulated in the bigger query I didn't post :) ).
I think it would be a good thing to add "parent" to the reserved word list in http://orientdb.com/docs/3.0.x/sql/SQL-Syntax.html#reserved-words

Thank you
Alessandro

**Comment by adepase** [comment]
Obviously you can close this for me

**Comment by luigidellaquila** [comment]
Hi @adepase 

Actually I think it's a bug ;-)
`$parent` is used to reference the parent query, but `parent` should be allowed as a normal property/projection name. I'm checking it, probably it's an easy fix

Thanks

Luigi

**Comment by adepase** [comment]
Anyway, please let us use "ancestor" at least (I don't like underscores and I lack synonyms) :)

**Comment by luigidellaquila** [comment]
Hi @adepase 

I pushed a fix, it will be released with v 3.0.11

Thanks

Luigi
